workers/generic-worker/os_groups_multiuser.go

```diff
@@ -19,7 +19,7 @@ func (osGroups *OSGroups) Start() *CommandExecutionError {
 	if len(groupNames) == 0 {
 		return nil
 	}
-	if config.RunTasksAsCurrentUser {
+	if osGroups.Task.Payload.Features.RunTaskAsCurrentUser {
```

Comment:
I suspect we should still add the task user to the groups, even if running as current user, as the docs say that the task user is still created, but the process is executed as current user. But this can be for a different PR.